### ISSUE TITLE
Question about the BED file

### ISSUE THREAD

**Issue by francicco** [hillerlab/TOGA#74]
Hi,

This is just a question. I was wondering what the colors in the `query_annotation.bed` file mean. Fusion? Paralogy?
Thanks a lot
F

**Comment by francicco** [comment]
Thank you @MichaelHiller,

I've looked at the supplementary. In it you characterize 5 states:


<img width="897" alt="Screenshot 2023-06-23 at 19 05 10" src="https://github.com/hillerlab/TOGA/assets/9006870/77037fbc-55dd-440b-b095-73fa84e4410d">

But I have 6 colors in my bed file. There's also
<img width="404" alt="Screenshot 2023-06-23 at 19 05 52" src="https://github.com/hillerlab/TOGA/assets/9006870/5f4d9430-723a-49f3-8fd9-d79a55e2a3c5">

They look like this in my annotation:
<img width="978" alt="Screenshot 2023-06-23 at 19 07 59" src="https://github.com/hillerlab/TOGA/assets/9006870/e92fe9e3-197d-47fe-9912-c76f4e488425">

Sorry for being so pedantic : )
Cheers
F

**Comment by MichaelHiller** [comment]
Good question and thanks for asking that. 

This is a special case called "paralogous projection". 
"A special and rarely occurring category called
“paralogous projection” refers to cases inwhich
no orthologous chain, but only a paralogousclassified
chain, was detected. This can arise if
the real orthologous gene is entirelymissing in
the assembly (thus only a paralog aligns) or if
TOGA misclassifies the orthologous gene because
of excessive divergence of intronic or intergenic
regions. If the locus represented by
the paralogous chain does not receive any annotation
through an orthologous chain, then
TOGA also annotates a gene at this locus
(shown in fig. S6), because this locus likely
encodes a gene. However, the annotation is labeled as a paralogous projection and is shown
in brown in fig. S6."

If you have many of these cases, you likely have evolutionary distances that are too large for TOGA. 

And yes, you likely have a fusion gene in your input. Fusion genes can be real or artifacts. 
If these 'brown' genes would have been recognized as orthologs, then TOGA would infer the wrong orthology relationships here (likely many:many).

**Comment by francicco** [comment]
Thank you Michael!

Because I have annotations for target species, I'm writing a series of scripts trying to use TOGA to infer orthology in order to infer orthology over my annotations and new genes that TOGA finds. So at the moment, the script I'm making finds all weird stuff.
I hope I don't have many of these instances because they look quite hard to solve.

Not sure if I'm interpreting this correctly, but what it looks like is that in the reference there are 4 copies (G1, G2, G3, G4: for genes in succession), of different lengths, while in the target, there is just one paralogous, therefore all map there. I'd interpret it as all belonging to the same orthologous group together with the reds.
Correct?
F

**Comment by MichaelHiller** [comment]
With target, you mean query species, I guess ? (I like reference / query as more precise terminology :-))

If I understand it right, you have annotations (likely transcriptome based) for all species and use TOGA to infer orthology relationships. Makes sense to me. 
If you have many paralogous projections, your evolutionary distances are likely too large (e.g. for human-mouse, we only get a few). 

"4 copies (G1, G2, G3, G4: for genes in succession)," --> You refer to the screenshot above, right? 
The answer depends on whether the different transcripts in the reference are annotated to be part of 1 gene (then the 5 exon transcript would be full length and the others are maybe incomplete annotations or shorter splice variants) or belong to 2 different genes (then the 5 exon transcript would be a fusion). 
If all transcripts would belong to 1 gene and if all projections would be classified as orthologous, TOGA would then use 'same-strand coding exon overlap' (see Fig S24) and infer for this locus that all 6 transcripts overlap and this locus has 1 gene. 
If the 5 exon transcript and the 2 / 3 shorter transcripts are in total annotated in the ref as 3 separate genes, then TOGA would infer that the 3 different genes all project to one locus, where they overlap. This would result in a 3:1 orthology relationship.

**Comment by francicco** [comment]
Yes, sorry reference / query indeed! And I think your description makes total sense.

In this two species, ~12 Mya divergence there are:
11280 Intact
2228 uncertain loss
1657 lost
529 missing
356 paralogous projection
110 partially intact

F

**Comment by MichaelHiller** [comment]
12 My is not much. Which species are we talking about? 

The summary looks alright, although there are a lot of genes with inact mutations. 
Of course, genes in the reference may not be ancestral and TOGA may infer a loss. Real gene losses can only be inferred if you have an outgroup with an intact gene. 
Alternatively, if most of these genes are ancestral, that either indicates fast evolution or an excess of base errors.

**Comment by francicco** [comment]
These are butterflies. The assembly was corrected with high-coverage genomic and RNAseq illumina data... Can't exclude errors... I tend to think more at a biological signal, the query genome seems to have a lot more genes, and the gene gain/loss turnover rate seems to be relatively high. But yes, can't exclude. I have to compare these numbers across different species, although I guess I have to normalize over the phylogenetic distances. Dividing the total mapped genes would work?
F
